flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/state/PendingSplitsStateSerializer.java

```diff
@@ -96,22 +97,46 @@ public int getVersion() {
 
     @Override
     public PendingSplitsState deserialize(int version, byte[] serialized) throws IOException {
-        if (version == VERSION) {
-            final DataInputDeserializer in = new DataInputDeserializer(serialized);
-            final int splitVersion = in.readInt();
-            final int stateFlag = in.readInt();
-            if (stateFlag == SNAPSHOT_PENDING_SPLITS_STATE_FLAG) {
-                return deserializeSnapshotPendingSplitsState(splitVersion, in);
-            } else if (stateFlag == BINLOG_PENDING_SPLITS_STATE_FLAG) {
-                return deserializeBinlogPendingSplitsState(in);
-            } else if (stateFlag == HYBRID_PENDING_SPLITS_STATE_FLAG) {
-                return deserializeHybridPendingSplitsState(splitVersion, in);
-            } else {
-                throw new IOException(
-                        "Unsupported to deserialize PendingSplitsState flag: " + stateFlag);
-            }
+        switch (version) {
+            case 1:
+                return deserializeV1(serialized);
+            case 2:
+                return deserializeV2(serialized);
+            default:
+                throw new IOException("Unknown version: " + version);
+        }
+    }
+
+    public PendingSplitsState deserializeV1(byte[] serialized) throws IOException {
+        final DataInputDeserializer in = new DataInputDeserializer(serialized);
+        final int splitVersion = in.readInt();
+        final int stateFlag = in.readInt();
+        if (stateFlag == SNAPSHOT_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeSnapshotPendingSplitsState(1, splitVersion, in);
+        } else if (stateFlag == BINLOG_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeBinlogPendingSplitsState(in);
+        } else if (stateFlag == HYBRID_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeHybridPendingSplitsState(1, splitVersion, in);
+        } else {
+            throw new IOException(
+                    "Unsupported to deserialize PendingSplitsState flag: " + stateFlag);
+        }
+    }
+
+    public PendingSplitsState deserializeV2(byte[] serialized) throws IOException {
+        final DataInputDeserializer in = new DataInputDeserializer(serialized);
+        final int splitVersion = in.readInt();
+        final int stateFlag = in.readInt();
+        if (stateFlag == SNAPSHOT_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeSnapshotPendingSplitsState(2, splitVersion, in);
+        } else if (stateFlag == BINLOG_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeBinlogPendingSplitsState(in);
+        } else if (stateFlag == HYBRID_PENDING_SPLITS_STATE_FLAG) {
+            return deserializeHybridPendingSplitsState(2, splitVersion, in);
+        } else {
+            throw new IOException(
+                    "Unsupported to deserialize PendingSplitsState flag: " + stateFlag);
```

Comment:
There is no much difference between `deserializeV2` and `deserializeV1`. Why not combine them and make version number as a parameter?